src/indra_cogex/sources/wikidata/__init__.py

```diff
@@ -0,0 +1,307 @@
+                # One row per relation:
+                # nlm_id, publisher_isni
+                if (nlm_id, isni) not in nlm_isni_relations:
+                    relations_writer.writerow((nlm_id, isni))
+                    nlm_isni_relations.add((nlm_id, isni))
+
+                # Save publishers
+                # One row per Publisher:
+                # publisher_wd_id, publisher_name, publisher_isni
+                if isni not in used_isni:
+                    publisher_writer.writerow([
+                        journal_publisher.publisher_wd_id,
+                        journal_publisher.publisher_name,
+                        isni
+                    ])
+                    used_isni.add(isni)
+
+                # Save journals
+                # One row per Journal:
+                # journal_wd_id, journal_name, issn_list, journal_issn_l, nlm_id
+                if journal_publisher.nlm_id not in used_nlm:
+                    journal_writer.writerow([
+                        journal_publisher.journal_wd_id,
+                        journal_publisher.journal_name,
+                        json.dumps(journal_publisher.journal_issn_list or []),
+                        journal_publisher.journal_issn_l,
+                        journal_publisher.nlm_id,
+                    ])
+                    used_nlm.add(journal_publisher.nlm_id)
+
+        logger.info(f"Dumped {len(used_isni)} publishers to "
+                    f"{self.publisher_data_path}")
+        logger.info(f"Dumped {len(used_nlm)} journals to "
+                    f"{self.journal_data_path}")
+        logger.info(f"Dumped {len(nlm_isni_relations)} relations to "
+                    f"{self.pub_jour_relations_data_path}")
+        logger.info(f"Skipped {skipped} relations due to missing NLM ID or "
+                    f"ISNI")
+
+    def get_nodes(self) -> Iterable[Node]:
+        """Get nodes from the data"""
+        self.process_data()
+
+        # Get journal and publisher nodes
+        yield from self._get_journal_nodes()
+        yield from self._get_publisher_nodes()
+
+    def _get_journal_nodes(self) -> Iterable[Node]:
+        with gzip.open(self.journal_data_path, 'rt') as fh:
+            reader = csv.reader(fh, delimiter='\t')
+            # journal_wd_id, journal_name, issn_list, journal_issn_l, nlm_id
+            for (
+                    journal_wd_id,
+                    journal_name,
+                    issn_list,
+                    journal_issn_l,
+                    nlm_id
+            ) in reader:
+                issn_list = json.loads(issn_list)
+                yield Node(
+                    "NLM",
+                    nlm_id,
+                    labels=[self.journal_node_type],
+                    data={
+                        "name": journal_name,
+                        "issn_l": journal_issn_l,
+                        "issn_list:string[]": ";".join(issn_list),
+                        "wd_id": "WD:" + journal_wd_id,
```

Comment:
Should we use or skip the `WD` prefix?

Reply:
skip it, this is just another banana situation

Reply:
Resolve in https://github.com/bgyori/indra_cogex/pull/139/commits/a31ca273877b526b4fb628b0073301343e698bf2